nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -874,24 +966,57 @@ contract StakingEscrow is Issuer, IERC900History {
     */
     function prolongStake(uint256 _index, uint16 _periods) external onlyStaker {
         StakerInfo storage info = stakerInfo[msg.sender];
-        require(_periods > 0, "Incorrect parameters");
+        // Incorrect parameters
+        require(_periods > 0);
         SubStakeInfo storage subStake = info.subStakes[_index];
         uint16 currentPeriod = getCurrentPeriod();
-        uint16 startPeriod = getStartPeriod(info, currentPeriod);
-        uint16 lastPeriod = getLastPeriodOfSubStake(subStake, startPeriod);
-        require(lastPeriod > currentPeriod, "The sub stake must active at least in the next period");
+        (uint16 startPeriod, uint16 lastPeriod) = checkLastPeriodOfSubStake(info, subStake, currentPeriod);
 
         subStake.periods = subStake.periods.add16(_periods);
         // if the sub stake ends in the next committed period then reset the `lastPeriod` field
         if (lastPeriod == startPeriod) {
             subStake.lastPeriod = 0;
         }
-        require(uint32(lastPeriod - currentPeriod) + _periods >= minLockedPeriods,
-            "The extended sub stake must not be less than the minimum value");
+        // The extended sub stake must not be less than the minimum value
+        require(uint32(lastPeriod - currentPeriod) + _periods >= minLockedPeriods);
         emit Locked(msg.sender, subStake.lockedValue, lastPeriod + 1, _periods);
         emit Prolonged(msg.sender, subStake.lockedValue, lastPeriod, _periods);
     }
 
+    /**
+    * @notice Merge two sub-stakes into one if their last periods are equal
+    * @dev It's possible that both sub-stakes will be active after this transaction.
+    * But only one of them will be active until next call `commitToNextPeriod` (in the next period)
+    * @param _index1 Index of the first sub-stake
+    * @param _index2 Index of the second sub-stake
+    */
+    function mergeStake(uint256 _index1, uint256 _index2) external onlyStaker {
+        StakerInfo storage info = stakerInfo[msg.sender];
+        SubStakeInfo storage subStake1 = info.subStakes[_index1];
+        SubStakeInfo storage subStake2 = info.subStakes[_index2];
+        uint16 currentPeriod = getCurrentPeriod();
+
+        (, uint16 lastPeriod1) = checkLastPeriodOfSubStake(info, subStake1, currentPeriod);
+        (, uint16 lastPeriod2) = checkLastPeriodOfSubStake(info, subStake2, currentPeriod);
+        // both sub-stakes must have equal last period to be mergeable
+        require(lastPeriod1 == lastPeriod2);
+        emit Merged(msg.sender, subStake1.lockedValue, subStake2.lockedValue, lastPeriod1);
```

Comment:
good question, it does not matter, because all code will executed as whole tx or fail as tx, and if we put it in the end - then need to store previous locked value which will consume some additional memory and gas